src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,93 @@
+package duke;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import duke.commands.*;
+import duke.tasks.*;
+
+/**
+ * The Parser class encapsulates the parsing of user input and file text when loading from the hard disk.
+ */
+public class Parser {
+    /**
+     * Parses the given input to determine the type of command the input contains.
+     * @param input Input to be parsed.
+     * @return Command object
+     * @throws DukeException
+     */
+    public static Command parseInput(String input) throws DukeException{
+        String[] splitInput = input.split(" ", 2);
+        String command = splitInput[0].trim();
+        String detail = "";
+        Boolean description_needed = command.equals("todo") || command.equals("deadline") || command.equals("event") ||
+                command.equals("mark") || command.equals("unmark") || command.equals("delete") || command.equals("find");
+        if (splitInput.length <= 1 && description_needed) {
+            throw new DukeException("Description of command is required.");
+        } else if (description_needed){
+            detail = splitInput[1].trim();
+        }
+
+        switch (command) {
+        case "list":
+            return new ListCommand();
+        case "mark":
+            return new MarkCommand(detail);
+        case "unmark":
+            return new UnmarkCommand(detail);
+        case "todo":
+            return new TodoCommand(detail);
+        case "deadline":
+            return new DeadlineCommand(detail);
+        case "event":
+            return new EventCommand(detail);
+        case "delete":
+            return new DeleteCommand(detail);
+        case "find":
+            return new FindCommand(detail);
+        default:
+            throw new DukeException(command + " is an invalid command.");
+        }
+    }
+
+    /**
+     * Parses text from file when tasks are loaded from the hard disk.
+     * @param line String to be parsed.
+     * @return Task parsed from input string.
+     */
+    public static Task parseFromFile(String line) {
+        String[] line_split = line.split("]", 3);
+        String type = line_split[0];
+        String status = line_split[1];
+        String rest = line_split[2];
```

Comment:
Uncaught `ArrayIndexOutOfBoundException`. 
While you did specify a length of 3, however, if the line is say `Test ] test`, it will not actually give you an array of length 3!
Perhaps you can add an assertion or check to make sure that the array is of length 3 before proceeding.
You might argue it is an save file, but there are always chances for the file to become corrupt or modified so I think this is something you should take note.